pkg/vault/util.go

```diff
@@ -98,6 +98,9 @@ func ReadVaultSecret(client Client, path, kvVersion string) (map[string]interfac
 		if _, ok := secret.Data["data"]; ok {
 			return secret.Data["data"].(map[string]interface{}), nil
 		}
+		if len(secret.Data) == 0 {
```

Comment:
Originally I had this check in the `Template.Replace()` method but the error handling here will swallow this case which merits a specific warning message, thus adding it here